src/parser/esprima_cpp/esprima.cpp

```diff
@@ -418,22 +418,37 @@ class Parser {
 
         VectorWithInlineStorage<8, SyntaxNode, std::allocator<SyntaxNode>>& params = paramsResult.params;
         bool hasParameterOtherThanIdentifier = false;
+        bool seenDefaultParameter = false;
 
         for (size_t i = 0; i < params.size(); i++) {
             switch (params[i]->type()) {
-            case Identifier: {
-                if (!hasParameterOtherThanIdentifier) {
+            case Identifier:
+                if (LIKELY(!seenDefaultParameter)) {
                     this->currentScopeContext->m_functionLength++;
                 }
                 break;
-            }
             case AssignmentPattern:
+                seenDefaultParameter = true;
+                hasParameterOtherThanIdentifier = true;
+                if (LIKELY(!seenDefaultParameter)) {
```

Comment:
This if statement seems useless because `seenDefaultParameter` value is always true here.

Reply:
Good point